lua/null-ls/builtins/diagnostics/cspell.lua

```diff
@@ -8,7 +8,13 @@ return h.make_builtin({
     filetypes = {},
     generator_opts = {
         command = "cspell",
-        args = { "stdin" },
+        args = function(params)
+            return {
+                "--language-id",
+                vim.api.nvim_buf_get_option(params.bufnr, "filetype"),
```

Comment:
```suggestion
                params.ft,
```

A small optimization. Looks good to me otherwise!

Reply:
Awesome, thanks @jose-elias-alvarez!